### DISCUSSION TITLE
Duplicate Key Violation in Supabase Due to Orphaned Records in pgsodium Schema

### DISCUSSION THREAD

**Discussion by SamGynius** [supabase/supabase#30603] [Questions] [answered]
Hello community,

I am currently facing a challenge with Supabase related to duplicate key violations when trying to insert records into a specific table. After executing a delete operation on several entries (which I did from the supabase dashboard manually), I discovered that the corresponding records still exist in the pgsodium schema, which is affecting my ability to perform new inserts.

I found lingering entries in the following tables within pgsodium:

key
mask_columns
masking_rule
valid_key

The insertion logic in my application utilizes an RPC function that generates keys in the pgsodium schema. However, because those deleted entries still have remnants within this read-only schema, I am encountering duplicate key violations for my new insertions.

I would like to know:

Is there a recommended approach to manage or clean up orphaned records in the pgsodium schema when they persist after a delete operation?
What steps should I take to prevent these errors from hindering my data insertion process, given that the schema is protected?

Thank you for any insights or guidance you can provide!

**Comment by GaryAustin1** [comment, marked as answer]
The schema and tables would still be accessible with the SQL editor.  I don't know enough about those tables and risks of deleting rows directly but you can certainly do it with the SQL editor.

pgsodium though is pending deprecation and not recommended.   You might want to make sure you understand https://supabase.com/docs/guides/database/extensions/pgsodium rational behind that.

You would also need to sort out why deleting in your main table is not cleaning up pgsodium tables if you plan to keep using it.

**Reply by SamGynius** [reply]
Thanks @GaryAustin1, I was able to do what I had to do in the SQL editor. Appreciate it!
